src/libraries/System.Private.CoreLib/src/System/ParseNumbers.cs

```diff
@@ -633,25 +633,14 @@ private static int GrabInts(int radix, ReadOnlySpan<char> s, ref int i, bool isU
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool IsDigit(char c, int radix, out int result)
         {
-            int tmp;
-            if ((uint)(c - '0') <= 9)
-            {
-                result = tmp = c - '0';
-            }
-            else if ((uint)(c - 'A') <= 'Z' - 'A')
-            {
-                result = tmp = c - 'A' + 10;
-            }
-            else if ((uint)(c - 'a') <= 'z' - 'a')
-            {
-                result = tmp = c - 'a' + 10;
-            }
-            else
+            int tmp = HexConverter.FromChar(c);
```

Comment:
Probably outside the scope of this PR, but we should probably change the `FromChar` helper to return -1 instead of 255 on failure. It would also simplify lots of consuming code since they could just check "< 0" and take advantage of all the magic bit-twiddling that comes along with it. I can send a PR for it unless you want to take it.